test/e2e/azure_lb.go

```diff
@@ -154,19 +174,37 @@ func AzureLBSpec(ctx context.Context, inputGetter func() AzureLBSpecInput) {
 		if !input.SkipCleanup {
 			By("deleting the ilb test resources")
 			Logf("deleting the ilb service: %s", ilbService.Name)
-			err = servicesClient.Delete(ctx, ilbService.Name, metav1.DeleteOptions{})
-			Expect(err).NotTo(HaveOccurred())
+			Eventually(func() error {
+				err := servicesClient.Delete(ctx, ilbService.Name, metav1.DeleteOptions{})
```

Comment:
based on our previous conversation in slack, my understanding was that this Delete doesn't actually wait for the service to be deleted, it just waits for the client go Delete to return, which doesn't fully wait for the service to be gone. Is that correct?

If so, waiting 20 minutes doesn't seem appropriate. We should change it to actually wait for the service to be gone so that we avoid running into flakes later due to cloud provider being in the middle of reconciling the deleted service when we try to create a new service.

Reply:
It's actually not clear (to me) in the client-go code if the delete operation blocks on success here...

https://github.com/kubernetes/client-go/blob/v0.23.0/kubernetes/typed/core/v1/service.go#L160

I don't see a "wait" or "do not wait" option here

https://github.com/kubernetes/apimachinery/blob/v0.23.0/pkg/apis/meta/v1/types.go#L482

Reply:
Based on the timing in logs it seems very likely that the delete is not blocking:

```
Mar  9 20:30:50.686: INFO: job default/curl-to-ilb-joboupqs is complete, took 10.069781536s
[1mSTEP[0m: deleting the ilb test resources
Mar  9 20:30:50.686: INFO: deleting the ilb service: webg7snfx-ilb
Mar  9 20:30:50.741: INFO: deleting the ilb job: curl-to-ilb-joboupqs
[1mSTEP[0m: creating an external Load Balancer service
Mar  9 20:30:50.776: INFO: starting to create an external Load Balancer service
```

We should add a check to make sure the service is gone before we proceed. We don't have to fix it in this PR but let's at least reduce the timeout, 20 minutes is too long if it's just for the action of starting the delete (and not the delete actually completing).